process_report/processors/coldfront_fetch_processor.py

```diff
@@ -0,0 +1,106 @@
+import os
+import sys
+import functools
+import logging
+from dataclasses import dataclass
+
+import requests
+
+from process_report.invoices import invoice
+from process_report.processors import processor
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+CF_ATTR_ALLOCATED_PROJECT_NAME = "Allocated Project Name"
+
+
+@dataclass
+class ColdfrontFetchProcessor(processor.Processor):
+    nonbillable_projects: list[str]
+
+    @functools.cached_property
+    def coldfront_client(self):
+        keycloak_url = os.environ.get("KEYCLOAK_URL", "https://keycloak.mss.mghpcc.org")
```

Comment:
I'm not suggesting that you change this now, but in general a good design principle is get configuration from the environment *outside* of the class (e.g., in your `main` method), and then pass that information into the class as class parameters. This makes things easier to test (you can just instantiate classes with your choice of values), and leads to better reuse (what if want to get values from a configuration file rather than the environment?).